src/modules/robot/Robot.cpp

```diff
@@ -1340,31 +1340,45 @@ bool Robot::append_milestone(const float target[], float rate_mm_s)
         // actual distance moved for this actuator
         // NOTE for a rotary axis this will be degrees turned not distance
         float d = fabsf(actuator_pos[actuator] - actuators[actuator]->get_last_milestone());
-        if(d == 0 || !actuators[actuator]->is_selected()) continue; // no movement for this actuator
+        if(d < 0.00001F || !actuators[actuator]->is_selected()) continue; // no perceptible movement for this actuator
 
         // find approximate min time this axis needs to move its distance
         float actuator_min_time= d / actuators[actuator]->get_max_rate();
         if (secs < actuator_min_time ) {
             // this move at the default rate will move too quickly for this actuator
             // so we decrease the overall feed rate so it can complete within the min time for this actuator
-            rate_mm_s= distance / actuator_min_time;
-            // recalculate time from new rate
-            secs = distance / rate_mm_s;
+            if(actuator >= N_PRIMARY_AXIS && !actuators[actuator]->is_extruder()) {
+                // if this is a rotary axis we check what the time would be if the given feedrate was
+                // for a solo move in degrees/sec so we do not go faster than that
+                // this is for rotary engraving where the feedrate is calculated to maintain
+                // a certain speed over the surface.
+                actuator_min_time= std::max(actuator_min_time, d/rate_mm_s);
```

Comment:
Hi @wolfmanjm 

am I interpreting this correctly? you want to drive the actuator outside of its specs i.e. above `actuators[actuator]->get_max_rate()` to keep up the feedrate that the G code requested?

I'm not sure that is a good idea. The requested G code feedrate can always just be a goal. It is always subject to machine constraints. If the actuator has to accelerate first, the feedrate will equaly not be met at first. 

However all moves are coordinated, so the axes will always be in sync to each other. If the rotation axis limits the overall toolpath feedrate, then the other axes are slowed to accomodate it. 

If a minmum feedrate is critical for the engraving (material melts otherwise?) the user must upgrade the rotation motor or change the engraving pattern. Stalling the motor won't help the engraving result either :-)

_Mark

Reply:
no misinterpreted, but irrelevant now anyway